File: app/src/main/java/com/example/kotlin/java/Child.java
package com.example.kotlin.java;

import android.os.Handler;
import android.os.Looper;
import android.os.Message;

import androidx.annotation.NonNull;

/**
 * @author zhangquan
 */
public class Child extends Father {

    @MyAnnotation
    private Handler handler2;

    public Child(String name) {
        super(name);
    }

    public Child(String name, int age) {
        super(name, age);
        handler2 = new Handler(Looper.getMainLooper());
    }

    public Child(String name, int age, long birth) {
        super(name, age);
    }

    @Override
    public void method() {
        value = 2;
        pubVar = "child pubvar";
    }

    @Override
    @MyAnnotation
    public String method2(String name) {
        handler.sendEmptyMessage(1);
        return name;
    }

    Handler handler = new Handler(Looper.getMainLooper()) {
        @Override
        public void handleMessage(@NonNull Message msg) {

        }
    };

}

File: app/src/main/java/com/example/kotlin/kotlin/Child.kt
package com.example.kotlin.kotlin

import android.os.Handler
import android.os.Looper
import android.os.Message
import com.example.kotlin.kotlin.annotation.MyAnnotation

/**
 * @author zhangquan
 */
class Child : Father {
    @MyAnnotation
    private var handler2: Handler? = null

    constructor(name: String?) : super(name!!) {}
    constructor(name: String?, age: Int) : super(name!!, age) {
        handler2 = Handler(Looper.getMainLooper())
    }

    constructor(name: String?, age: Int, birth: Long) : super(name!!, age) {}

    override fun method() {
//        value = 2;
//        pubVar = "child pubvar";
    }

    @MyAnnotation
    override fun method2(name: String): String {
        handler.sendEmptyMessage(1)
        return name
    }

    var handler: Handler = object : Handler(Looper.getMainLooper()) {
        override fun handleMessage(msg: Message) {}
    }
}